vespajlib/src/main/java/ai/vespa/validation/Hostname.java

```diff
@@ -0,0 +1,30 @@
+// Copyright Yahoo. Licensed under the terms of the Apache 2.0 license. See LICENSE in the project root.
+package ai.vespa.validation;
+
+import java.util.regex.Pattern;
+
+import static ai.vespa.validation.Validation.requireMatch;
+
+/**
+ * A valid hostname.
+ *
+ * @author jonmv
+ */
+public class Hostname extends StringWrapper<Hostname> {
+
+    public static final Pattern hostnameLabel = Pattern.compile("([A-Za-z0-9]|[A-Za-z0-9][A-Za-z0-9-]{0,61}[A-Za-z0-9])");
+    public static final Pattern hostnamePattern = Pattern.compile("(?=.{1,255})(" + hostnameLabel + "\\.)*" + hostnameLabel);
```

Comment:
In practice, hostnames are limited to 64 characters. At least on Linux.

Reply:
Hmm ... so there's a difference in "host part of a URI" and a "hostname", I guess. I used the former definition. Perhaps the latter is significantly different?